src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/IncrementingEventCounter.cs

```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+#if ES_BUILD_PCL
+    using System.Threading.Tasks;
+#endif
+
+#if ES_BUILD_STANDALONE
+namespace Microsoft.Diagnostics.Tracing
+#else
+namespace System.Diagnostics.Tracing
+#endif
+{
+    /// <summary>
+    /// IncrementingEventCounter is a variant of EventCounter for variables that are ever-increasing. 
+    /// Ex) # of exceptions in the runtime.
+    /// It does not calculate statistics like mean, standard deviation, etc. because it only accumulates
+    /// the counter value.
+    /// </summary>
+    internal partial class IncrementingEventCounter : BaseCounter
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IncrementingEventCounter"/> class.
+        /// IncrementingEventCounter live as long as the EventSource that they are attached to unless they are
+        /// explicitly Disposed.   
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <param name="eventSource">The event source.</param>
+        public IncrementingEventCounter(string name, EventSource eventSource) : base(name, eventSource)
+        {
+            _increment = 0;
```

Comment:
Nit: its no harm, but you don't need this. CLI guarantees fields are zero/default initialized.